STYLEGUIDE.md

```diff
@@ -0,0 +1,221 @@
+```md
+<Package overview>
+## Table of contents
+* [classes](_csharp-api/packagename/classes/)
+* [childclasses](_csharp-api/packagename/childclasses/)
+* ...
+## Classes
+* [classname](_csharp-api/packagename/classes/classname/)
+* ...
+## Child Classes
+* [classname](_csharp-api/packagename/childclasses/classname/)
+* ...
+## Template Classes
+* [classname](_csharp-api/packagename/templateclasses/classname/)
+* ...
+## Enums
+* [enumname](_csharp-api/packagename/enums/enumname/)
+* ...
+## Structs
+* [structname](_csharp-api/packagename/structs/structname/)
+* ...
+## Interfaces
+* [interfacename](_csharp-api/packagename/interfaces/interfacename/)
+* ...
+## Exception Classes
+* [exceptionname](_csharp-api/packagename/exceptions/exceptionname/)
+* ...
+```
+>Each of the sections (at both levels) should have a valid link to their respective info pages.
+
+The package information header should contain the following:
+```md
+Current Version (Stable branch): v0.0.0.0
+<Brief description of the package and its contents>
+```
+___
+
+## Classes (Base, Child, Templated, and Exceptions)
+
+Unlike the root and package level READMEs, the class README will contain more than just links to child folders. Each class will start with a **Class Overview Panel**:
+
+### [Parent Object Name](EXAMPLE)
+# Class Name
+
+This is a description of the class and what it does.
+
+<div style= "font-size:12px">
+
+| Using / Namespaces | <div style='color:yellow'>Taleworlds.Core</div> | <div style='color:green'>Saveable</div> |
+|:---|---|---:|
+| [System](EXAMPLE) | | 10000 |
+| [System.Xml](EXAMPLE)| | |
+| [TaleWorlds.SaveSystem](EXAMPLE) | | |
+</div>
+
+> For the non-savable classes, use ```'color:red'``` and change to text to "Not Savable".
+
+After this class overview, a small code snippet should exist showing how to use / initialize the class. **This should not be exhaustive**, just a quick reference.
+```csharp
+new InformationMessage("Hello World!")
+```
+
+The above would be a valid snippet for the ```InformationMessage``` Class, but can be longer if the extra length is used to give important context about the snippet.
+
+Directly after this intro snippet, we have a section outlining the variables of the class. All variables are assumed to have private setters unless otherwise noted. Each variable should contain all of the following information:
+
+* Name
+* Type (with link to that object or objects)
+* Notes
+
+## Variables
+
+<div style= "font-size:12px">
+
+| Mod | Name | Type | Notes |
+| --- | --- | --- | --- |
+| Const | HeadArmor | int |
+| Instance |MeshesMask | [SkinMask](EXAMPLE) | |
+| ReadOnly |BeardCoverType | [ArmorComponent](EXAMPLE).[BeardCoverTypes](EXAMPLE)
+| Static |ReinsRopeMesh | string | returns ```this.ReinsMesh + "_rope";```|
+|...|...|...|...|
+</div>
+
+Methods should follow parameters, seperated by a H1 header.
+
+## Methods
+
+<div style= "font-size:12px">
+
+| Type | Method Name | Number of Params |
```

Comment:
Number of Params - redundant?

Reply:
Params better, I think.

Reply:
Agreed. Overthought it while writing. Fixed.